pytorch_lightning/callbacks/base.py

```diff
@@ -46,6 +46,30 @@ def on_sanity_check_end(self, trainer, pl_module):
         """Called when the validation sanity check ends."""
         pass
 
+    def on_train_epoch_start(self, trainer, pl_module):
+        """Called when the train epoch begins."""
+        pass
+
+    def on_train_epoch_end(self, trainer, pl_module):
+        """Called when the train epoch ends."""
+        pass
+
+    def on_val_epoch_start(self, trainer, pl_module):
```

Comment:
we named it "validation" in the callbacks below. It would be great if we could make it consistent.